spectrum_io/search_result/openms.py

```diff
@@ -0,0 +1,164 @@
+import pandas as pd
+
+from pyopenms import IdXMLFile
+from spectrum_fundamentals.mod_string import openms_to_internal
+from tqdm import tqdm
+from spectrum_io.search_result.search_results import SearchResults, filter_valid_prosit_sequences
+
+
+def strToFloat(df):
+  for col in df:
+    try:
+      df[col] = [float(i) for i in df[col]]
+    except ValueError:
+      continue
+  return df
+
+def readAndProcessIdXML(input_file, top=1):
```

Comment:
@picciama should also get everything in idXML not just top 1 hit?

Reply:
This is referring to the number of hits considered per spectra? Oktoberfest is using only the top hit at the moment, this is default. So this is a feature for the future that also requires a change in Oktoberfest, i.e. a config parameter that determines how many hits to consider, but this has to be done for other search engines as well. There is actually an issue present in Oktoberfest for exactly that: https://github.com/wilhelm-lab/oktoberfest/issues/118

Reply:
Agreed.